### ISSUE TITLE
Raster Merge produces negative Dimension

### ISSUE THREAD

**Issue by GGDRriedel** [qgis/QGIS#37315]
The Merging Function produces a negative dimension, as already posted in the previous version

Error Protocol: 

> QGIS-Version: 3.10.2-A Coruña
QGIS-Codeversion: 616ad4531b
Qt-Version: 5.11.2
GDAL-Version: 3.0.2
GEOS-Version: 3.8.0-CAPI-1.13.1 
PROJ-Version: Rel. 6.2.1, November 1st, 2019
Verarbeite Algorithmus…
Algorithmus Verschmelzen startet…
Input parameters:
{ 'DATA_TYPE' : 5, 'EXTRA' : '', 'INPUT' : ['C:/Users/Riedel/Projekte/fun/wandern/334445638_dgm1.xyz','C:/Users/Riedel/Projekte/fun/wandern/334445640_dgm1.xyz','C:/Users/Riedel/Projekte/fun/wandern/334465636_dgm1.xyz','C:/Users/Riedel/Projekte/fun/wandern/334465638_dgm1.xyz','C:/Users/Riedel/Projekte/fun/wandern/334465640_dgm1.xyz','C:/Users/Riedel/Projekte/fun/wandern/334485638_dgm1.xyz'], 'NODATA_INPUT' : None, 'NODATA_OUTPUT' : None, 'OPTIONS' : '', 'OUTPUT' : 'C:/Users/Riedel/Projekte/fun/wandern/test.tif', 'PCT' : False, 'SEPARATE' : False }

GDAL command:
python3 -m gdal_merge -ot Float32 -of GTiff -o C:/Users/Riedel/Projekte/fun/wandern/test.tif --optfile C:\Users\Riedel\AppData\Local\Temp/processing_7c6cecac1d6744d6a5c2bb3a8d321471/4ee4855bf4a845aa9964efafe07f5beb/mergeInputFiles.txt
GDAL command output:
ERROR 1: Attempt to create 6000x-1999 dataset is illegal,sizes must be larger than zero.

Creation failed, terminating gdal_merge.

Execution completed in 18.71 seconds
Results:
{'OUTPUT': 'C:/Users/Riedel/Projekte/fun/wandern/test.tif'}

Lade Ergebnis Layer
Die folgenden Layer wurden nicht erzeugt.<ul><li>C:/Users/Riedel/Projekte/fun/wandern/test.tif</li></ul>Im 'Protokoll-Fenster' im QGIS-Hauptfenster sind mehr Informationen zur Ausführung des Algorithmus zu finden.

_Originally posted by @GGDRriedel in https://github.com/qgis/QGIS/issues/34284#issuecomment-646559409_

**Comment by GGDRriedel** [comment]
This also happens in 3.13

QGIS-Version: 3.13.0-Master
QGIS-Codeversion: e9a0c01b21
Qt-Version: 5.11.2
GDAL-Version: 3.2.0dev
GEOS-Version: 3.8.1-CAPI-1.13.3
PROJ-Version: Rel. 7.1.0, August 1st, 2020
Verarbeite Algorithmus…
Algorithmus Verschmelzen startet…
Input parameters:
{ 'DATA_TYPE' : 5, 'EXTRA' : '', 'INPUT' : ['C:/Users/Riedel/Projekte/fun/wandern/334465638_dgm1.xyz','C:/Users/Riedel/Projekte/fun/wandern/334465640_dgm1.xyz'], 'NODATA_INPUT' : None, 'NODATA_OUTPUT' : None, 'OPTIONS' : '', 'OUTPUT' : 'C:/Users/Riedel/Projekte/fun/wandern/test.tif', 'PCT' : False, 'SEPARATE' : False }

GDAL command:
python3 -m gdal_merge -ot Float32 -of GTiff -o C:/Users/Riedel/Projekte/fun/wandern/test.tif --optfile C:/Users/Riedel/AppData/Local/Temp/processing_bPQABH/bd2dbcac4c624151a63fb66c583dbba9/mergeInputFiles.txt
GDAL command output:
ERROR 1: Attempt to create 2000x0 dataset is illegal,sizes must be larger than zero.

Creation failed, terminating gdal_merge.

Execution completed in 7.07 seconds
Results:
{'OUTPUT': 'C:/Users/Riedel/Projekte/fun/wandern/test.tif'}

Lade Ergebnis Layer
Die folgenden Layer wurden nicht erzeugt.
• C:/Users/Riedel/Projekte/fun/wandern/test.tif
Im 'Protokoll-Fenster' im QGIS-Hauptfenster sind mehr Informationen zur Ausführung des Algorithmus zu finden.

**Comment by GGDRriedel** [comment]
Does ONLY happen with .xyz ASCII raster files so far

**Comment by gioman** [comment]
> Does ONLY happen with .xyz ASCII raster files so far

@GGDRriedel does it happens if you run the (GDAL) command from the command line? if it does then is likely an issue in GDAL. Please test.

**Comment by GGDRriedel** [comment]
It seems this is a GDAL problem and can be found all over the internet: 

https://gis.stackexchange.com/questions/142754/gdal-merge-tries-to-create-image-with-negative-dimensions

It was also reported as "already fixed" here: 
https://github.com/qgis/QGIS/issues/34284


But it isn't. 

I tested some more, happens with converted .tifs as well

**Comment by gioman** [comment]
> It was also reported as "already fixed" here:
> #34284
> 
> But it isn't.

@GGDRriedel (regarding GDAL bases Processing tools) QGIS is "just" a GDAL gui, if there is an issue in GDAL then this will affect also QGIS.

If there as a fix in GDAL you must check in what version this fix was added/backported and check if the QGIS version you are using runs with a GDAL version that has the fix.

**Comment by GGDRriedel** [comment]
I tried tracing this down to GDAL and it turns out it is due to my inouts seemingly having a negative pixel resolution, which throws off GDAL. 
Their resolution is shown as 1,-1 in properties. 

What I ended up with to fix this without going through issue reporting  hell is converting all of them to single geotiffs WITH VRT generation and merging all of those. 

it's incredibly numb but at least it works, I needed this to join DEM tiles
